rfcs/00x-frontend-library-split.md

```diff
@@ -0,0 +1,88 @@
+
+### Allow for multiple frontends
+
+Extracting the API server into its own applications would allow us to create different frontends for the Amundsen application. Think about creating a mobile application, a desktop app or forking the current UI application to customize it deeply to the user's specifications.
+
+It would also make it easier for other applications to query Amundsen, integrating its results in other applications.
+
+### Streamlined Development
+
+Splitting the frontend repository would make it easier to develop in both systems, as updates and issues on one application won't affect the other. For example, we just had an issue where we couldn't test a change in the frontend app because a dependency on the backend app was incorrect.
+
+Reducing the complexity of the project will improve developer's focus and create a simpler, more straightforward development experience.
+
+## Guide-level Explanation (aka Product Details)
+
+For a final user of Amundsen, there shouldn't be any change. This is an architectural-level refactor.
+
+## UI/UX-level Explanation
+
+No changes should be seen in the UI level, but this change would open doors for new and exciting evolutions on performance for the UI and the API responses.
+
+## Reference-level Explanation (aka Technical Details)
+
+Technically, the high level changes would be:
+
+- Create a new repository for the frontend, probably based on a ready-made application like Next.js
+- Move our frontend application into this new system, updating the endpoints and wiring to use the current frontend implementation
+- Remove the frontend application from our current frontend repository
+
+## Drawbacks
+
+Why we shouldn't do this?
+
+- Load of work involved
+- Migration costs for current users
+
+## Alternatives
+
+We consider this the best approach as it lines up with general best practices, decouples our system and adds flexibility while reducing complexity.
+
+One alternative would be to do this same work while keeping both in the same repository. It won't bring all the benefits, but it will be a bit better than the current status.
+
+Not doing this would further leave us behind the latests improvements on the field, will continue having a complex setup and onboarding, making it hard to contribute to and hard to extend our platform.
+
+## Prior art
+
+We have talked several times about this, never tried it. It has been a common move in all internal projects at Lyft though.
+
+## Unresolved questions
+
+We'll need to create a detailed step-by-step plan for this refactor.
+
+## Future possibilities
+
+For the backend:
+
+- Easily open the backend to other services and applications
+
+For the frontend:
+
+- Update it to use a static site generator like Gatsby
+- Update it to become a Next.js application, incorporating best practices and best performance
```

Comment:
why next.js?

Reply:
Just an example, but is it one of the most popular platforms right now. They help by setting a lot of defaults out of the box, making it easier to setup a new application. It also provides a good amount of interesting features from the start.